File: BotsSDK/korebot/src/main/java/kore/botssdk/adapter/AdvanceTableListOuterAdapter.java
package kore.botssdk.adapter;

import android.content.Context;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

import kore.botssdk.R;
import kore.botssdk.models.AdvanceListTableModel;
import kore.botssdk.utils.KaFontUtils;

public class AdvanceTableListOuterAdapter extends BaseAdapter
{
    private final Context context;
    private final ArrayList<AdvanceListTableModel> arrAdvanceListTableModels;

    public AdvanceTableListOuterAdapter(@NonNull Context context, @NonNull ArrayList<AdvanceListTableModel> arrAdvanceListTableModels)
    {
        this.context = context;
        this.arrAdvanceListTableModels = arrAdvanceListTableModels;
    }

    @Override
    public int getCount()
    {
        return arrAdvanceListTableModels.size();
    }

    @Override
    public Object getItem(int i) {
        return arrAdvanceListTableModels.get(i);
    }

    @Override
    public long getItemId(int i) {
        return i;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup viewGroup) {
        DetailsViewHolder holder;
        if (convertView == null) {
            convertView = View.inflate(context, R.layout.advance_list_table_outer_cell, null);
            KaFontUtils.applyCustomFont(context, convertView);
            holder = new DetailsViewHolder();
            holder.rvTableList = convertView.findViewById(R.id.rvTableList);
            convertView.setTag(holder);
        } else {
            holder = (DetailsViewHolder) convertView.getTag();
        }

        holder.rvTableList.setLayoutManager(new GridLayoutManager(context, 2));
        holder.rvTableList.setAdapter(new AdvanceTableListAdapter(context, arrAdvanceListTableModels.get(position).getRowData()));

        return convertView;
    }

    static class DetailsViewHolder {
        RecyclerView rvTableList;
    }
}

File: BotSdk/ui/src/main/java/com/kore/ui/adapters/AdvanceTableListOuterAdapter.kt
package com.kore.ui.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.kore.model.constants.BotResponseConstants
import com.kore.ui.R

class AdvanceTableListOuterAdapter(
    val context: Context,
    val buttons: List<Map<String, *>>,
) : RecyclerView.Adapter<TableViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): TableViewHolder {
        return TableViewHolder(LayoutInflater.from(context).inflate(R.layout.advance_list_table_outer_cell, parent, false))
    }

    override fun getItemCount(): Int {
        return buttons.size
    }

    override fun onBindViewHolder(holder: TableViewHolder, position: Int) {
        val button = buttons[position]
        holder.rvTableList.layoutManager = GridLayoutManager(context, 2)
        holder.rvTableList.adapter = AdvanceTableListAdapter(context, button[BotResponseConstants.ROW_DATA] as List<Map<String, *>>)
    }

}

class TableViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
    var rvTableList: RecyclerView

    init {
        rvTableList = itemView.findViewById(R.id.rvTableList)
    }
}